### DISCUSSION TITLE
Nightscout on Google Cloud does not receive treatments from AAPS. NSClient error: io.socket.engineio.client.EngineIOException: xhr poll error

### DISCUSSION THREAD

**Discussion by dar-kod** [NightscoutFoundation/xDrip#2358] [General]
AAPS 3.1.0.3

Nightscout is installed on Google Cloud following instructions at https://navid200.github.io/xDrip/docs/Nightscout/GoogleCloud.html

In AAPS -> OVERVIEW there is 1 message:

`NSClient malfunction. Consider NS and NSClient restart.`

And in AAPS -> NSCLIENT there is this error:

`io.socket.engineio.client.EngineIOException: xhr poll error`

In NS I created one profile and in AAPS also one with different name, but they do not sync. Treatments, basal rates... from AAPS are not uploaded in NS.

**Comment by Navid200** [comment]
Do you see your readings in Nightscout from xDrip?
If you use a browser, can you see your Nightscout?

**Reply by dar-kod** [reply]
Yes, readings are visible in Nightscout from xDrip+.
Yes, I can see Nightscout in a browser.

**Reply by Navid200** [reply]
Only for troubleshooting, change denied to admin.  
I have updated this page to include instructions:  
https://navid200.github.io/xDrip/docs/Nightscout/NS_Variables  

But, that is an extremely unsafe setting and should be changed back as soon as possible.

**Reply by dar-kod** [reply]
With this change, it is the same.

I forgot to write, SMBs from AAPS were received at the beginning of using Nightscout, in the first 3 hours, but this NSClient error was also present.

**Comment by jamorham** [comment]
Which android version is NSClient running on?

**Reply by dar-kod** [reply]
MIUI 12.5 -> Android 11

**Reply by Navid200** [reply]
@dar-kod There is a new release.
Unfortunately, there is no way to just update.  It requires deleting everything even the virtual machine and starting everything from scratch.
What's worse is that none of us has tested it!  Your setup is kind of special.  
So, even if you go ahead and start from scratch, I cannot guarantee that the problem you are experiencing will be resolved.

If you like to give it a try, please go ahead and delete everything and start from scratch and when you are done, please tell us if it solved the problem or not.

**Reply by dar-kod** [reply]
@Navid200 Do you mean new release of Nightscout, and it is not possible to update it in the terminal?
The Nightscout data is only a few days old, so there is no problem with deleting it. Do you mean to just delete the VM Nightscout instance and with that everything will be deleted, or I have to delete something else? Will readings from xDrip be sent to AAPS during that time of the update?

**Reply by Navid200** [reply]
While re-installing Nightscout, you will have no Nightscout.  So, it will not be able to send anything to AAPS.

Can you switch to your Heroku Nightscout before doing this repair?

It is an update to our Nightscout setup on Google Cloud.  It is not an update to Nightscout.
If you want to try it, you will need to switch to your Heroku Nightscout first so that your control is not disturbed.
Then, you will go to the compute engine.
You will click on the 3-dot menu on the right end of the line showing the virtual machine and select Stop.
After it stops, you will use the same menu to delete.
Now, you will go through all the steps of creating a new virtual machine making sure to follow all the details to keep it free, and installing Nightscout.
But, when you do that, the line you paste into the terminal to install will do it differently (update).

**Reply by dar-kod** [reply]
It's not my Nightscout. I will set up a new Heroku Nightscout first.

**Comment by Atlas-Night-Out** [comment]
Figured the solutions will document it later Thanks! Its to do with Settings with AAPS causing it all!

**Reply by dar-kod** [reply]
@Atlas-Night-Out Thank you very much! Where will it be documented?

@Navid200 I do not need to do an update to Nightscout setup on Google Cloud?

**Reply by Navid200** [reply]
@dar-kod 
I already told you what you should do.
These are growing pains.
We are still testing everything.
And we are grateful that you have tested and informed us of the problem you have experienced.

I have no idea what @Atlas-Night-Out means and what he is getting at by his post in this thread.


The installation procedure has been updated.  Unfortunately, none of us has the setup nsclient.  So, we cannot reproduce the problem you are experiencing.
So, even after the update, it may not solve your problem.

If you want to give it a try, you need to follow the instructions I gave you already.
I am going to provide them again hoping that it will be more clear this time:

You will lose all readings that have been uploaded to Nightscout on Google Cloud.
While you are going through the update procedure, Nightscout on Google Cloud will not be active.

1- You need to stop the virtual machine.  To do that, you need to go to the dashboard and click on Compute Engine.  Then, click on the 3-dot menu on the virtual machine line and select Stop.  Wait until this is completed.  it may take a minutes or 2.

2- You need to delete the virtual machine.  To do that, click on the 3-dot menu button on the virtual machine line on the compute engine page and select delete.
Wait for this to complete.

3- You need to set up Nightscout on Google Cloud from scratch.  To do that, follow all the instructions (1-6) here:  https://navid200.github.io/xDrip/docs/Nightscout/GoogleCloud

**Reply by dar-kod** [reply]
@Navid200 You wrote: "You need to set up Nightscout on Google Cloud from scratch. To do that, follow all the instructions (1-6)"

Do you mean to follow the instructions (3-6) as the project is already created, and you didn't write to delete the project?

**Reply by Navid200** [reply]
Yes, you are correct.  It's 3 to 6.

**Reply by dar-kod** [reply]
Now Nightscout on Google Cloud works with AAPS. How can I change “Config Vars” with Ubuntu terminal?

**Comment by Navid200** [comment]
Everything is here:  https://navid200.github.io/xDrip/docs/Nightscout/GoogleCloud
 
If you scroll down on that page, you will see items other than the steps required to install.
For example, you can see "Nightscout variables".
If you click on that link, it will take you to a page, I have explained how you can edit Nightscout variables.

If it is not clear, please let me know.

**Reply by dar-kod** [reply]
I want to add basal rates. In Heroku -> Settings -> Config Vars, there was BASAL_RENDER="default". Following the logic, I added this line at the end:

`export BASAL_RENDER="default"`

But still basal rates are not visible.

**Reply by Navid200** [reply]
Did you reboot the virtual machine after editing the nsconfig file?

**Reply by dar-kod** [reply]
I forgot to reboot. It's fine now. Thank you.

**Comment by andrewfaid** [comment]
I was also struggling with this.  I deleted my Google Nghtscout and rebuilt it with the updated instructions and it worked!!  I did get a few xhr poll errors when I reconnected, but the Nightscout Queue cleared and everything is looking good.
Thanks for the help

**Comment by spanosp** [comment]
Hi everyone!

First of all many thanks to @Navid200 and anyone else involved on the NS GCS guide!

I installed NS on GCS yesterday and the treatments do not show up, I can only see the glucose readings. The treatments are shown as expected on my Heroku NS. Also, the db % is lower on the GCS NS (7%) than on my Heroku NS (11%).

Any suggestions?

Thank you!

**Reply by jonprogers** [reply]
@spanosp I think I noticed something similar which I reported here; https://github.com/NightscoutFoundation/xDrip/discussions/2299#discussioncomment-3842822

When I tried to upload to two ns instances from xDrip (heroku and google cloud) by adding two BaseURLs in the xDrip settings, the treatments only got uploaded to the first one listed. I don't know where looking into this is on priority scale of those looking into such things but in the interim I know to list the ns instance with my main database first...

Have I understood you correctly - Does this mirror what you are seeing?

**Reply by spanosp** [reply]
@jonprogers That was it!! thank you

silly question, if using GCS, is the mongodb obsolete?

**Reply by Navid200** [reply]
We are still using MongoDB.
That's why you can backup and restore your database using MongoDB commands:
https://navid200.github.io/xDrip/docs/Nightscout/DatabaseBackup.html
https://navid200.github.io/xDrip/docs/Nightscout/DatabaseRestore.html

But, your MongoDB database is stored on your Google Cloud virtual disk.

**Reply by Navid200** [reply]
@spanosp @jonprogers If you swap the position of the two URLs in the Base URL setting on the xDrip Nightscout Sync page, do the treatments always upload to the first one on the list?

Or, is it only the Nightscout on the Cloud that fails to receive treatments regardless of the order of the two URLs in that setting?

**Reply by jonprogers** [reply]
They always go to the first one in the list, I've swapped them backwards and forwards a few times and the treatments always go to the first one listed, whichever one that is.

I've also upgraded to a later nightly (15/10/22) since my original post and the behavior is the same.

Let me know if I can provide any further info...

**Reply by Navid200** [reply]
@jonprogers There is an existing issue that seems to be related:  https://github.com/NightscoutFoundation/xDrip/issues/266

Would you please post in that issue and explain that you are experiencing the same issue with treatments?
It will be very important to reveal that blood glucose readings are successfully uploaded to both.  But, not treatments.  It is again very useful to reveal that the treatments are successfully uploaded to the first URL regardless of which one is placed first.

Thanks

**Reply by spanosp** [reply]
@jonprogers @Navid200 same for me, swapping the urls around works, treatments are only sent to the 1st url

**Reply by jonprogers** [reply]
@Navid200 posted in https://github.com/NightscoutFoundation/xDrip/issues/266#issuecomment-1290761527 as requested

**Comment by TiagoPRSilva** [comment]
Hello everyone, sorry to be a bit late to the conversation. Is there any news regarding the reason behind this issue ? I have nightscout selfhosted for some time and after my last visit to my doctor i noticed the treatments missing in my NS and the same error in AAPS NSClient. i looked at my nginx error and access logs and no evidence of the issue, so i think it must be at the ns itself. Does anyone in the same scenario (self hosted) have the same issue ? i will try to diagnose myself, if i find anything ill share. Thanks

**Reply by Navid200** [reply]
@TiagoPRSilva 
The person who opened this discussion, stated in this same discussions that everything was working for him.

I suggest that you open a new discussion and we can go over your setup.
The fact that you are not aware the situation was resolved suggests that this particular discussion has become too long and you did not read all the posts in it.
Please open a new discussion.

**Reply by TiagoPRSilva** [reply]
Ok, thanks
